server/src/queues/unlinkDatabase.ts

```diff
@@ -0,0 +1,80 @@
+import { dbTypeToDokkuPlugin } from './../graphql/utils';
+import { Database, App } from '../generated/graphql';
+import { Worker, Queue } from 'bullmq';
+import createDebug from 'debug';
+import { pubsub } from './../index';
+
+import { config } from '../config';
+import { sshConnect } from '../lib/ssh';
+import { prisma } from '../prisma';
+
+const queueName = 'unlink-database';
+const debug = createDebug(`queue:${queueName}`);
+
+interface QueueArgs {
+  app: App;
+  database: Database;
+}
+
+export const unlinkDatabaseQueue = new Queue<QueueArgs>(queueName, {
+  defaultJobOptions: {
+    // Max timeout 20 minutes
+    timeout: 1.2e6,
+  },
+  connection: config.redisClient,
+});
+
+/**
+ * - Unlink db
+ */
+const worker = new Worker(
+  queueName,
+  async (job) => {
+    const { app, database } = job.data;
+    debug(
+      `starting unlinkDatabaseQueue for ${database.type} database ${database.name} from  ${app.name} app`
+    );
+    console.log(
+      `starting unlinkDatabaseQueue for ${database.type} database ${database.name} from  ${app.name} app`
+    );
+
+    const dbType = dbTypeToDokkuPlugin(database.type);
+
+    const ssh = await sshConnect();
+
+    const res = await ssh.execCommand(
+      `${dbType}:unlink ${database.name} ${app.name}`
+    );
+
+    const arrayOfLogs = res.stdout.split('\n');
+
+    pubsub.publish('DATABASE_UNLINKED', { unlinkDatabaseLogs: arrayOfLogs });
```

Comment:
the problem of sending the logs at the end of the exec command is that if the task is taking 10 mins, we will only get the log in the UI 10 mins after. We should rather stream them to have them in real time.

Reply:
@pradel yeah, this one I was not really able to figure out and currently we are sending one bulk of text 😬 .